src/model/constraints.testdata.ts

```diff
@@ -0,0 +1,552 @@
+import { readonlyRecord } from 'fp-ts';
+
+import { Constraint } from './system/core';
+import { Syntax } from './system/expander';
+
+export type DecodeTest = {
+  value: string
+  expected: Constraint
+}
+
+export const decodeTests: readonlyRecord.ReadonlyRecord<string, DecodeTest> = {
+  ////////////  HCP tests  //////////////////////
+  // Make sure plus operator works as expected
+  hcpPlusSyntax:
+  {
+    value: "7+",
+    expected:
+    {
+      "type": "PointRange",
+      "min": 7,
+      "max": 37
+    }
+  },
+
+  // Make sure minus operator works as expected
+  hcpMinusSyntax:
+  {
+    value: "7-",
+    expected:
+    {
+      "type": "PointRange",
+      "min": 0,
+      "max": 7
+    }
+  },
+
+  ////////////////  Suit length tests  //////////////////////
+  // Suit length equality operator
+  suitEqualitySyntax:
+  {
+    value: "5=H",
```

Comment:
Instead of having a single test for the constant 5, generate this number from the appropriate range, construct the `value` given the generated number, construct the appropriate `expected` given the generated number, and then assert actual equals expected.

Reply:
This is all very new to me.  So far, so good.   However, I'm more interested in getting everything up and running initially, even if I have hard coded some numbers.  I am leery of getting lost in the weeds too soon without focusing on hitting some testable milestones.

Reply:
That is the correct mindset. Keep up the great work!